src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+
+    static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo).append(System.lineSeparator());
+
+        try {
+            for (int i = 0; i < names.length; ++i) {
+                int totalSalary = 0;
+                for (String record : data) {
+                    totalSalary += calculateSalaryForNameAndDate(names[i],
+                            record, dateFrom, dateTo);
+                }
+                result.append(names[i]).append(" - ").append(totalSalary);
+
+                if (i != names.length - 1) {
```

Comment:
Dont get why this is required, is it possible to add a line break after you add a row?